Adding test scripts (YAMLs) for Laundry Dryer Mode

prakashece opened the pull request:
The test scripts are based on LaundryDryerControls.adoc. Please find the link below to access it.

https://github.com/CHIP-Specifications/chip-test-plans/pull/3686

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=30716) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=30716) before we can accept your contribution.<br/><hr/>**Prakash  Ravi** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=30716) it.</sub>

woody-apple commented:
@prakashece Please check restyle, it's complaining about the format of the YAML document.

prakashece commented:
> @prakashece Please check restyle, it's complaining about the format of the YAML document.

@woody-apple Fixed the restyle issue.

manjunath-grl reviewed (changes requested):
@prakashece The script should begin with the inclusion of CHIP copyrights.
https://github.com/project-chip/connectedhomeip/blob/fb47c905f96e348976612105faa5ef04654e8888/src/app/tests/suites/certification/Test_TC_MOD_3_2.yaml#L1

manjunath-grl commented:
Test case IDs should adhere to the format, for example: Test_TC_<TestcaseName>_ID.yaml. Additionally, automated test cases must be included in [ciTests.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/ciTests.json)

prakashece commented:
> Test case IDs should adhere to the format, for example: `Test_TC_<TestName>_ID.yaml`. Additionally, automated test cases must be included in [ciTests.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/ciTests.json)

@manjunath-grl addressed your comments. ciTests.json will be pushed in another Pull request

raju-apple reviewed (changes requested):
1. @prakashece Why do we have 2 copies of the YAMLs , can we delete the other one not adhering to the format ? 
2. Please add the new tests also in the ciTests.json 
Here's a sample PR for reference https://github.com/project-chip/connectedhomeip/pull/28985

prakashece commented:
> 1. @prakashece Why do we have 2 copies of the YAMLs , can we delete the other one not adhering to the format ?
> 2. Please add the new tests also in the ciTests.json
>    Here's a sample PR for reference [Modified tests Aug 31 #28985](https://github.com/project-chip/connectedhomeip/pull/28985)

@raju-apple Deleted the other 2 files and added the tests in ciTests.json. Please check and let me know if you have any comments

manjunath-grl commented on src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:
```diff
@@ -0,0 +1,97 @@
+    cluster: "Laundry Dryer Controls"
+    endpoint: 1
+
+unsupportedSelectedDrynessLevel:
```
Here Indentation is wrong.

prakashece commented on src/app/tests/suites/certification/TC-DRYERCTRL-1.1.yml:
```diff
@@ -0,0 +1,77 @@
+          constraints:
+              type: int16u
+
+    - label: "Step 3: Read the global attribute: FeatureMap"
```
@manjunath-grl updated the test scripts

cecille commented on src/app/tests/suites/certification/PICS.yaml:
```diff
@@ -8877,6 +8877,53 @@ PICS:
+    - label: "Does the DUT support testing the failed ChangeToMode command?"
+      id: DRYERCTRL.S.M.CAN_TEST_MODE_FAILURE
+
+    - label:
```
I don't think this file is meant to have pixits in it.

prakashece replied:
Removed the Pixits

cecille commented on src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:
```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021 Project CHIP Authors
```
```suggestion
# Copyright (c) 2023 Project CHIP Authors
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRYERCTRL_1_1.yaml:
```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2021 Project CHIP Authors
```
```suggestion
# Copyright (c) 2023 Project CHIP Authors
```

cecille commented on src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:
```diff
@@ -0,0 +1,102 @@
+    cluster: "Laundry Dryer Controls"
+    endpoint: 1
+
+    unsupportedSelectedDrynessLevel:
```
I know this is more of a test plan thing, but is the device guaranteed to HAVE an unsupported selected dryness level?

prakashece replied:
It doesn't have an unsupported dryness level it is just for the test case

Just to check if it doesn't accept an unsupported dryness level

cecille commented on src/app/tests/suites/certification/ci-pics-values:
```diff
@@ -750,6 +750,25 @@ PIXIT.LWM.MODE_CHANGE_OK=1
+DRYERCTRL.S.C00.Rsp=1
+DRYERCTRL.S.C01.Tx=1
+
+#PIXIT
```
These aren't required here.

prakashece replied:
updated

prakashece commented:
Created a new one https://github.com/project-chip/connectedhomeip/pull/31250
